src/org/zaproxy/zap/extension/alertFilters/AlertFilterTableModel.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.alertFilters;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.view.AbstractMultipleOptionsTableModel;
+
+/**
+ * A table model for holding a set of AlertFilter, for a {@link Context}.
+ */
+public class AlertFilterTableModel extends AbstractMultipleOptionsTableModel<AlertFilter> {
+
+	/** The Constant defining the table column names. */
+	private static final String[] COLUMN_NAMES = {
+			Constant.messages.getString("alertFilters.table.header.enabled"),
+			Constant.messages.getString("alertFilters.table.header.alertid"),
+			Constant.messages.getString("alertFilters.table.header.url"),
+			Constant.messages.getString("alertFilters.table.header.newalert") };
+
+	/** The Constant serialVersionUID. */
+	private static final long serialVersionUID = 4463944219657112162L;
+
+	/** The alert filters. */
+	private List<AlertFilter> alertFilters = new ArrayList<>();
```

Comment:
Consider removing the initialisation, it's not necessary, both constructors are already doing it.